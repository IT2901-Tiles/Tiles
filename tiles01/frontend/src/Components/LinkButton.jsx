import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import '../CSS/Button.css';
import Button from 'react-bootstrap/Button';

//Button component 
//props:
//title: the title on the button
//target: tha page the button is going to direct to
//size: aize of the button, must be Large or Small
//category: which category the button us linking to, optional
//category will also change the colour of the button
//options for categoryare: trigger, thing, feedback, animation

function LinkButton(props) {
    const [title] = useState(props.title)
    var [path] = useState(props.target)
    const [category] = useState(props.category)
    const [size] = useState(props.size + "Button")
    const [active,setActive] = useState(props.active);
    let colour = "#F08A00";
    let fontColour = "#F3EAC2";


    if (category === "things") {
        colour = "#D64539"
    } else if (category === "feedback") {
        colour = "#FFCD00"
        fontColour = "#484848"
    } else if (category === "animation") {
        colour = "#84AD64"
    }

    if( props.disabled === true){
        return <Button style={{ backgroundColor: colour, color: fontColour }} disabled={true} className={size} variant="primary"  >{title}</Button>
    }
    

    
    
    
    return (
        <Link
        to={{
            pathname: path,
            state: { category: category, color: colour, card: props.card }
            // state: { category: category, color:colour }
        }} >
            <Button style={{ backgroundColor: colour, color: fontColour }} className={size} variant="primary" onClick={props.onClick} >{title}</Button>
        </Link>
    );
}




export default LinkButton;
