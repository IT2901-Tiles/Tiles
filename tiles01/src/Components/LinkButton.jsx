import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import './Button.css';
import Button from 'react-bootstrap/Button';

//Button component 
//props:
//title: the title on the button
//target: tha page the button is going to direct to
//size: aize of the button, must be Large or Small
//category: which category the button us linking to, optional
//category will also change the colour of the button
//options for categoryare: trigger, thing, feedback, animation

function LinkButton(props) {
    const [title, setTitle] = useState(props.title)
    const [path, setPath] = useState(props.target)
    const [category, setCategory] = useState(props.category)
    const [size, setSize] = useState(props.size + "Button")

    let colour = "#F08A00"
    let fontColour = "#F3EAC2"

    if (category === "thing") {
        colour = "#D64539"
    } else if (category === "feedback") {
        colour = "#FFCD00"
        fontColour = "#484848"
    } else if (category === "animation") {
        colour = "#84AD64"
    }

    return (
        <Link
            to={{
                pathname: path,
                state: { category: category, card:props.card }
            }}>
            <Button style={{ backgroundColor: colour, color: fontColour }} className={size} variant="primary">{title}</Button>
        </Link>
    );
}

export default LinkButton;
