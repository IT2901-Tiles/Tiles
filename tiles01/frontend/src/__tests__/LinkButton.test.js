import {configure, shallow} from "enzyme"
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {cleanup} from '@testing-library/react';
import LinkButton from '../Components/LinkButton';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from "react-router-dom";


afterEach(() => {
    cleanup();
});

configure({
    adapter: new Adapter()
});

describe('LinkButton', () => {
    const mock = jest.fn()
    const wrapper = shallow(<LinkButton onClick={mock} category="things"/>)
    it('renders correctly', () => {
        const tree = renderer.create(<Router><LinkButton/></Router>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('contains a link element', () => {
        expect(wrapper.find('Link')).toHaveLength(1);
    });
    it('contains a button element', () => {
        expect(wrapper.find('Button')).toHaveLength(1);
    });
    it('Button triggers function', () => {
        wrapper.find('Button').simulate('click')
        expect(mock).toHaveBeenCalled()
    });
    it('Right color is chosen, based on category prop, things', () => {
        expect(wrapper.find('Link').props().to.state.color).toEqual("#D64539");
    });
    it('Right color is chosen, based on category prop, animation', () => {
        const colorWrapper = shallow(<LinkButton onClick={mock} category="animation"/>)
        expect(colorWrapper.find('Link').props().to.state.color).toEqual("#84AD64");
    });
});