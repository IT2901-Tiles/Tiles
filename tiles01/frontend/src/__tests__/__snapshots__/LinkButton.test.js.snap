// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton renders correctly 1`] = `
<a
  href="/"
  onClick={[Function]}
>
  <button
    className="undefinedButton btn btn-primary"
    disabled={false}
    style={
      Object {
        "backgroundColor": "#F08A00",
        "color": "#F3EAC2",
      }
    }
    type="button"
  />
</a>
`;
